zebra-consensus/src/script.rs

```diff
@@ -12,11 +12,181 @@
+
+                async move {
+                    let previous_output = match output.await? {
+                        zebra_state::Response::Utxo(output) => output,
+                        _ => unreachable!("AwaitUtxo always responds with Utxo"),
+                    };
+
+                    zebra_script::is_valid(
+                        transaction,
+                        branch_id,
+                        (input_index as u32, previous_output),
+                    )?;
+
+                    Ok(())
+                }
+                .boxed()
+            }
+            transparent::Input::Coinbase { .. } => unimplemented!(
+                "how should we handle verifying coinbase transactions in the script::Verifier?"
+            ),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
+
+    use color_eyre::Report;
+
+    use zebra_chain::{
+        block::Block, parameters::Network, parameters::NetworkUpgrade,
+        serialization::ZcashDeserializeInto, transparent,
+    };
+    use zebra_test::transcript::{TransError, Transcript};
+
+    #[tokio::test]
+    async fn happy_path() -> Result<(), Report> {
+        happy_path_test().await
+    }
+
+    #[spandoc::spandoc]
+    async fn happy_path_test() -> Result<(), Report> {
+        zebra_test::init();
+        let mut inputs = zebra_test::vectors::MAINNET_BLOCKS
+            .range(1..10)
+            .flat_map(|(_, block_bytes)| {
+                let block: Block = block_bytes.zcash_deserialize_into().unwrap();
+                block.transactions.into_iter()
+            })
+            .map(|transaction| {
+                transaction
+                    .inputs()
+                    .iter()
+                    .cloned()
+                    .collect::<Vec<transparent::Input>>()
+            })
+            .flat_map(|inputs| inputs.into_iter());
+
+        assert!(inputs.any(|input| matches!(input, transparent::Input::PrevOut { .. })));
+
+        let state_setup_transcript =
+            zebra_test::vectors::MAINNET_BLOCKS
+                .range(0..=10)
+                .map(|(_, block_bytes)| {
+                    let block: Arc<Block> = block_bytes.zcash_deserialize_into().unwrap();
+                    let hash = block.hash();
+
+                    let request = zebra_state::Request::CommitFinalizedBlock { block };
+                    let response = Ok::<_, TransError>(zebra_state::Response::Committed(hash));
+
+                    (request, response)
+                });
+        let state_setup_transcript = Transcript::from(state_setup_transcript);
+
+        let config = zebra_state::Config::ephemeral();
+        let network = Network::Mainnet;
+
+        let state = zebra_state::init(config, network);
+
+        /// SPANDOC: Setup the state by commiting all the necessary context
+        state_setup_transcript.check(state.clone()).await?;
+
+        let branch = NetworkUpgrade::Overwinter.branch_id().unwrap();
```

Comment:
Blocks 1..10 are `BeforeOverwinter`, so we can't validate these scripts - they have no branch_id.

(Branch Ids were introduced in `Overwinter`.)